clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1304,6 +1310,19 @@ private CompletedFetch initializeCompletedFetch(CompletedFetch nextCompletedFetc
         return completedFetch;
     }
 
+    private void handleOffsetOutOfRange(long fetchOffset,
```

Comment:
I think we can let this take `FetchPosition` instead of just the offset. Inside `initializeCompletedFetch`, we can pull the position from this line:
```java
                    if (fetchOffset != subscriptions.position(tp).offset) {
```